fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/EnforceAndCostTask.java

```diff
@@ -271,10 +272,10 @@ private boolean canGenerateOneStageAgg(GroupExpression childBestExpr) {
         // 1. check the agg node is global aggregation without split and child expr is PhysicalDistributionOperator
         if (aggregate.getType().isGlobal() && !aggregate.isSplit() &&
                 childBestExpr.getOp() instanceof PhysicalDistributionOperator) {
-            // 2. check default column statistics or output row count less than 1
-            if ((groupExpression.getGroup().getStatistics().getColumnStatistics().values().stream()
+            // 2. check default column statistics or child output row may not be accurate
+            if (groupExpression.getGroup().getStatistics().getColumnStatistics().values().stream()
                     .allMatch(ColumnStatistic::isUnknown) ||
-                    groupExpression.getGroup().getStatistics().getOutputRowCount() <= 1)) {
+                    childBestExpr.getGroup().getStatistics().isTableRowCountMayNotAccurate()) {
```

Comment:
A new way to prevent the generation of one-stage aggregation, the previous judgment method is wrong, will be wrong to judge the limit 1 case. I also add some backgroud above.